sormas-app/app/src/main/java/de/symeda/sormas/app/immunization/edit/ImmunizationEditFragment.java

```diff
@@ -0,0 +1,220 @@
+		initialDistricts = InfrastructureDaoHelper.loadDistricts(record.getResponsibleRegion());
+		initialCommunities = InfrastructureDaoHelper.loadCommunities(record.getResponsibleDistrict());
+	}
+
+	@Override
+	protected void onLayoutBinding(FragmentImmunizationEditLayoutBinding contentBinding) {
+		contentBinding.setData(record);
+
+		contentBinding.setYesNoUnknownClass(YesNoUnknown.class);
+
+		contentBinding.immunizationCountry.initializeSpinner(countries);
+
+		InfrastructureDaoHelper.initializeRegionFields(
+			contentBinding.immunizationResponsibleRegion,
+			initialRegions,
+			record.getResponsibleRegion(),
+			contentBinding.immunizationResponsibleDistrict,
+			initialResponsibleDistricts,
+			record.getResponsibleDistrict(),
+			contentBinding.immunizationResponsibleCommunity,
+			initialResponsibleCommunities,
+			record.getResponsibleCommunity());
+
+		InfrastructureDaoHelper.initializeRegionFieldListeners(
+			contentBinding.immunizationResponsibleRegion,
+			contentBinding.immunizationResponsibleDistrict,
+			record.getResponsibleDistrict(),
+			contentBinding.immunizationResponsibleCommunity,
+			record.getResponsibleCommunity(),
+			null,
+			null,
+			null,
+			null,
+			null,
+			() -> false);
+	}
+
+	@Override
+	public void onAfterLayoutBinding(final FragmentImmunizationEditLayoutBinding contentBinding) {
+
+		// Initialize ControlSpinnerFields
+		contentBinding.immunizationDisease.initializeSpinner(diseaseList);
+		contentBinding.immunizationImmunizationStatus.initializeSpinner(immunizationStatusList);
+		contentBinding.immunizationImmunizationManagementStatus.initializeSpinner(immunizationManagementStatusList);
+		contentBinding.immunizationMeansOfImmunization.initializeSpinner(meansOfImmunizationList);
+
+		// Initialize ControlDateFields
+		contentBinding.immunizationReportDate.initializeDateField(getFragmentManager());
+		contentBinding.immunizationRecoveryDate.initializeDateField(getFragmentManager());
+		contentBinding.immunizationStartDate.initializeDateField(getFragmentManager());
+		contentBinding.immunizationEndDate.initializeDateField(getFragmentManager());
+		contentBinding.immunizationLastInfectionDate.initializeDateField(getFragmentManager());
+
+		contentBinding.immunizationMeansOfImmunization.addValueChangedListener(e -> {
+			if (e.getValue() == MeansOfImmunization.OTHER || e.getValue() == MeansOfImmunization.RECOVERY) {
+				contentBinding.immunizationImmunizationManagementStatus.setValue(ImmunizationManagementStatus.COMPLETED);
+				contentBinding.immunizationImmunizationManagementStatus.setEnabled(false);
+				contentBinding.overwriteImmunizationManagementStatusCheckBox.setVisibility(View.VISIBLE);
+			}
+			if (e.getValue() == MeansOfImmunization.VACCINATION || e.getValue() == MeansOfImmunization.VACCINATION_RECOVERY) {
+				contentBinding.immunizationVaccinationHeading.setVisibility(View.VISIBLE);
+				contentBinding.immunizationRecoveryHeading.setVisibility(View.VISIBLE);
```

Comment:
Recovery section and its fields are only supposed to be visible when RECOVERY or VACCINATION_RECOVERY is selected, not when VACCINATION is selected; I suggest to add surrounding linear layouts for the vaccination and the recovery fields and headings so that you only have to change its visibility and not the visibility of all fields that depend on this logic